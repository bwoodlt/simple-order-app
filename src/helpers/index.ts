export const firstLetterToUpper = (value:string):string => value[0].toUpperCase()+value.slice(1)
