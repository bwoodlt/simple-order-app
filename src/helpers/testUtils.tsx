import { render, RenderOptions, RenderResult } from '@testing-library/react';
import {  PropsWithChildren, ReactElement } from 'react';
import {createMemoryHistory}  from 'history'
import { MemoryRouter, Router } from 'react-router-dom';

const providerWrapper = ({children}: PropsWithChildren<unknown>): ReactElement => <MemoryRouter>{children}</MemoryRouter>

export const renderWrapper = (ui: ReactElement, options?: RenderOptions): RenderResult => {
    return render(ui, {wrapper: providerWrapper, ...options}, )
}

export const renderWithMemoryHistory = (ui: ReactElement, route = '/', history = createMemoryHistory({initialEntries: ['/']})) => ({...render(<Router history={history}>{ui}</Router>), history})
