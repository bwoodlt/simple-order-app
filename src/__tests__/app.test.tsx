import {
  render,
  fireEvent,
  waitFor,
  screen,
  RenderResult,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from '../app';
import { renderWithMemoryHistory, renderWrapper } from '../helpers/testUtils';

describe('App', () => {
  it('render App', () => {
    const {asFragment}: RenderResult = render(<App />);

    // output comparison -> prevents regression
    expect(asFragment).toMatchSnapshot();
  });

  it('Loads and shows Go to Order Page button', async () => {

    renderWrapper(<App />);

    const buttonElement = screen.getByText(/Go to Order Page/);
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement.firstChild?.nodeValue).toEqual('Go to Order Page');
  });
});
