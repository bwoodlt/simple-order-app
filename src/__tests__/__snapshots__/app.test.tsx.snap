// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render App 1`] = `[Function]`;
